staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -323,6 +329,12 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.WarningHeaders) {
+		for _, w := range crdInfo.warnings[requestInfo.APIVersion] {
+			warning.AddWarning(req.Context(), "", w)
```

Comment:
Dumb question: is it okay for `agent` to be empty here?

Reply:
yes, when we generate the header, an agent of "" is serialized as `-`. this matches how we produce the deprecated warnings for in-tree types